### PR TITLE
BATM-990 BATM SSH Tunnel Server

### PR DESCRIPTION
**Pull request by pvyhnal-generalbytes**
This is a standalone application running on a remote host which
implements SSH server and allows to make tunnelled (ssh encrypted)
connections to a wallet running on that host.

Deployment:
- Build and publish `batm_ssh_tunnel-1.0.0.tar`
- Edit URL variable at the beginning of the `batm_ssh_tunnel_install.sh` and publish that too

Installation:
- on a remote host where your wallet (e.g. bitcoind) is installed run:
 `curl --proto '=https' --tlsv1.2 -sSf https://hq.generalbytes.com/.../batm_ssh_tunnel_install.sh | sh`
- it outputs a configuration for a new field in CAS > Crypto Settings > View > Buy or Sell > Wallet Tunnel (added in a separate PR)

### CODE REVIEW

File: batm_ssh_tunnel/batm_ssh_tunnel_install.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+URL="file:///home/pvyhnal/git/batm_main/batm_public/batm_ssh_tunnel/build/distributions/batm_ssh_tunnel-1.0.0.tar" # TODO
```
Comment (pvyhnal-generalbytes) [inline batm_ssh_tunnel/batm_ssh_tunnel_install.sh]: Use publicly accessible HTTPS URL here

File: batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            throw new IOException("Cannot create " + CONFIG_DIR.getAbsolutePath());
+        }
+
+
+        if (args.length > 0 && "init".equals(args[0])) {
+            String password = new Auth(CONFIG_DIR).init();
+            System.out.println(password); // intentionally stdout and not logging. The output is used in the install script
+
+        } else {
+            int port = Integer.parseInt(System.getProperty("batmSshTunnelPort", "22222"));
```
Comment (generalbytes) [inline batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java]: do not use camelcase for property names. use _ if you need to separate words.

Comment (generalbytes) [reply batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java]: Move 22222 into constant

File: batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            new SshServer(CONFIG_DIR, port).start();
+
```
Comment (generalbytes) [inline batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java]: Write into console instructions for user what to do next. Go to CAS blabla...

Comment (pvyhnal-generalbytes) [reply batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java]: Instructions are displayed be the install script:

Adding user batmsshtunnel
Downloading and unpacking to /opt/batm_ssh_tunnel
Generating password
Installing the service
Created symlink /etc/systemd/system/multi-user.target.wants/batm-ssh-tunnel.service → /etc/systemd/system/batm-ssh-tunnel.service.
Starting the service

Please make the port 22222 is accessible from BATM Master Server.
No other ports need to be open.
Use this Tunnel Password in CAS Crypto Settings.
Make sure to keep the password SECRET!

********************************
Sh6f4MXb1&uRL%G2y3U-s%OR7EEIbJLq
********************************

File: batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java (hunk 2 above)
Comment (pvyhnal-generalbytes) [reply batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java]: done

**Pull request merged**